src/access_nri_intake/catalog/utils.py

```diff
@@ -26,3 +27,28 @@ def wrapper(*args, **kwargs):
         return _to_tuple(series)
 
     return wrapper
+
+
+def trace_failure(func: Callable) -> Callable:
+    """
+    Decorator that wraps a function and prints a message if it raises an exception
+    """
+    func_name = func.__name__
+    colname = func_name[1:].split("_")[0]
```

Comment:
Exactly what issue are you having with getting the function name out?

Reply:
`func.__name__` will return `wrapper` here - seems to be a common problem eg. https://stackoverflow.com/questions/50350379/getting-function-name-from-nested-decorator, but none of the solutions I've found seem to have solved it yet.

Reply:
Fixed by changing the order of decoration & moving definitions in - about to tidy up & push.

Reply:
Cool, if you take it out of Draft I can approve it so it can go in